File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        #i: string 
        #o: number represent the length of the longest substring without repeating charactoers
        #c: s consists of letter/digits/symbols and spaces
        #e: if input string length = 0 
        #handle the edge case
        if len(s) == 0:
            return 0
        res = 1
        left = 0
        right = 1

        while right < len(s):
            cur = s[left: right + 1]
            if len(set(cur)) == len(cur):
                if len(cur) > res:
                    res = len(cur)
                right += 1
            else:
                left += 1
                right = left + 1
        return res

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
 //i：string with characters no numbers
 //o: number present the length of the longest subsring
 //c: 
 //e: !s.length
 //"abbcd" -> bcd => 3
    if (!s.length) {
        return 0
    }
    let left = 0;
    let right = 1;
    let res = 1;
    while (right < s.length) {
        const temp = s.slice(left, right + 1)
        if (new Set(temp).size === temp.length) {
            res = Math.max(res, temp.length)
            right += 1
        } else {
            left += 1
            right = left + 1
        }
    }
    return res
};